src/lib/sqlite/schema.ts

```diff
@@ -0,0 +1,54 @@
+import { sqliteTable, text, integer, real, unique, primaryKey } from 'drizzle-orm/sqlite-core'
+import { sql } from 'drizzle-orm'
+
+export const snippets = sqliteTable(
+  'snippets',
+  {
+    id: text('id').primaryKey(),
+    folderId: text('folder_id').references(
+      () => folders.id,
+      {
+        onDelete: 'cascade',
+        onUpdate: 'cascade',
+      },
+    ),
+    name: text('name').notNull(),
+    language: text('language').notNull(),
+    text: text('text').notNull(),
+    encrypted: integer('encrypted', {mode: 'boolean'}).default(false).notNull(),
+    position: real('position').default(1).notNull(),
+    updatedAt: text('updated_at').default(sql`CURRENT_TIMESTAMP`).notNull(),
+    createdAt: text('created_at').default(sql`CURRENT_TIMESTAMP`).notNull(),
+  },
+)
+
+export const folders = sqliteTable(
+  'folders',
+  {
+    id: text('id').primaryKey(),
+    name: text('name').notNull(),
+    position: real('position').default(1).notNull(),
+    updatedAt: text('updated_at').default(sql`CURRENT_TIMESTAMP`).notNull(),
+    createdAt: text('created_at').default(sql`CURRENT_TIMESTAMP`).notNull(),
+  },
+)
```

Comment:
There is a fundamental difference between the implemented `Catalog` (or a map of existing folders), and the newly-implemented `folders`: going to a high-level view, one is a map, and the other is an array. The current code for different components is quite coupled with the map structure of `Catalog`. To make it works, we can 

- Try "molding" the array from the database to the map, or
- Rewrite the dependencies of the components; separate the functionalities to a "cleaner" API

The question is that should we follow the first or the second option? The first one is quite ugly, but can be "seemingly" fast to be implemented. The second will be a better design, but it would require more work.